From 8f615cda88ee5cd33b4ee9a1f76ba177956a10d8 Mon Sep 17 00:00:00 2001
From: Sai Krishna Juturi <jsaikrishna@codeaurora.org>
Date: Wed, 22 Nov 2017 16:40:18 +0530
Subject: net: usb: rmnet_usb_ctrl: Fix use after free issue

In rmnet_usb_ctrl_init dev is freed again it is
dereferenced with in same block. Fix this issue
by taking derefernced value into a local variable
prior to freeing dev.

Change-Id: Ic51accc8949e97c8aaeda558adcc9d404fb2a26d
Signed-off-by: Sai Krishna Juturi <jsaikrishna@codeaurora.org>
---
 drivers/net/usb/rmnet_usb_ctrl.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/usb/rmnet_usb_ctrl.c b/drivers/net/usb/rmnet_usb_ctrl.c
index 75e9783..25ebe136 100644
--- a/drivers/net/usb/rmnet_usb_ctrl.c
+++ b/drivers/net/usb/rmnet_usb_ctrl.c
@@ -1219,12 +1219,13 @@ skip_cudev_init:
 						     "%s%d", rmnet_dev_names[i],
 						     n);
 			if (IS_ERR(dev->devicep)) {
+				long status = PTR_ERR(dev->devicep);
 				pr_err("%s: device_create() returned %ld\n",
-					__func__, PTR_ERR(dev->devicep));
+					__func__, status);
 				cdev_del(&dev->cdev);
 				free_rmnet_ctrl_udev(dev->cudev);
 				kfree(dev);
-				return PTR_ERR(dev->devicep);
+				return status;
 			}
 
 			/*create /sys/class/hsicctl/hsicctlx/modem_wait*/
-- 
cgit v1.1

